src/v/s3/client.cc

```diff
@@ -515,7 +516,11 @@ client_pool::client_pool(
 
 ss::future<> client_pool::stop() {
     _as.request_abort();
-    return _gate.close();
+    _cvar = {}; // this should break concurrent 'acquire' loop that waits for
```

Comment:
what is this line doing? it looks like it doesn't do anything, but it looks like you're looking for something like _cvar.broken() behavior?

Reply:
Fixed that.